File: front/src/app/core/enums/estado-civil.enum.ts
const listaEstadoCivil: EstadoCivilEnum[] = [];

export class EstadoCivilEnum {
  static readonly SOLTEIRO = new EstadoCivilEnum(1, 'SOLTEIRO');
  static readonly CASADO = new EstadoCivilEnum(2, 'CASADO');
  static readonly DIVORCIADO = new EstadoCivilEnum(3, 'DIVORCIADO');
  static readonly VIUVO = new EstadoCivilEnum(4, 'VIÚVO');
  static readonly SEPARADO = new EstadoCivilEnum(5, 'SEPARADO');
  static readonly UNIAO_ESTAVEL = new EstadoCivilEnum(6, 'UNIÃO ESTÁVEL');

  private constructor(
    public readonly id: number,
    public readonly descricao: string
  ) {
    listaEstadoCivil.push(this);
  }

  public static getAllValues(): EstadoCivilEnum[] {
    return listaEstadoCivil;
  }

  public static getById(id: number): EstadoCivilEnum | undefined {
    return listaEstadoCivil.find((estadoCivil) => estadoCivil.id === id);
  }

  public static getByDescricao(descricao: string): EstadoCivilEnum | undefined {
    return listaEstadoCivil.find(
      (estadoCivil) => estadoCivil.descricao === descricao
    );
  }
}

File: back/src/main/java/web/api/br/formulario/enums/EstadoCivilEnum.java
package web.api.br.formulario.enums;

import com.fasterxml.jackson.annotation.JsonValue;

public enum EstadoCivilEnum {
    SOLTEIRO(1, "SOLTEIRO"),
    CASADO(2, "CASADO"),
    DIVORCIADO(3, "DIVORCIADO"),
    VIUVO(4, "VIÚVO"),
    SEPARADO(5, "SEPARADO"),
    UNIAO_ESTAVEL(6, "UNIÃO ESTÁVEL");

    private final Integer id;
    private final String descricao;

    EstadoCivilEnum(Integer id, String descricao) {
        this.id = id;
        this.descricao = descricao;
    }

    @JsonValue
    public Integer getId() {
        return id;
    }

    public String getDescricao() {
        return descricao;
    }

    public static EstadoCivilEnum getById(Integer id) {
        for (EstadoCivilEnum estadoCivil : values()) {
            if (estadoCivil.getId().equals(id)) {
                return estadoCivil;
            }
        }
        return null;
    }

    public static EstadoCivilEnum getByDescricao(String descricao) {
        for (EstadoCivilEnum estadoCivil : values()) {
            if (estadoCivil.getDescricao().equalsIgnoreCase(descricao)) {
                return estadoCivil;
            }
        }
        return null;
    }
}
